### PR TITLE
Workflow CRUD API with Standardized Error Handling

### PR DESCRIPTION
**Pull request by renancaju**
## Summary

This PR implements a complete workflow CRUD API with comprehensive error handling, validation, and testing. The implementation follows clean architecture principles and includes both unit and integration tests.

#### **Complete CRUD Operations**
- ✅ **CREATE** - `POST /workflows` - Create new workflows with validation
- ✅ **READ** - `GET /workflows` - List workflows with optional owner filtering  
- ✅ **READ** - `GET /workflows/:id` - Get workflow by ID
- ✅ **UPDATE** - `PATCH /workflows/:id` - Update workflows with partial update support
- ✅ **DELETE** - `DELETE /workflows/:id` - Delete workflows (soft delete)

#### **Advanced Features**
- ✅ **PUBLISH** - `POST /workflows/:id/publish` - Publish workflows using publishing service
- ✅ **DRAFT CREATION** - `POST /workflows/groups/:groupId/create-draft` - Create draft from published workflow
- ✅ **Owner Filtering** - `GET /workflows?owner_id=user123` - Filter workflows by owner
- ✅ **Request Validation** - Full validation with detailed error messages
- ✅ **RFC 7807 Compliance** - Proper error responses using Problem Details

#### **Repository Layer Enhancements**
- Extended `WorkflowRepository` interface with `GetAllByOwner()` method
- Implemented owner-based filtering in both PostgreSQL and file persistence layers
- Updated all mock implementations for testing compatibility

#### **Service Layer Updates**
- Enhanced `Workflow` service with `FetchAllByOwner()` method
- Leveraged existing `Publishing` service for status change operations
- Maintained clean separation of concerns

#### **Web/API Layer**
- Created comprehensive request/response DTOs with validation rules
- Implemented all CRUD handlers with proper error handling
- Enhanced existing `GetWorkflows` handler with owner filtering
- Added status change handlers for publish and draft operations

#### **Dependency Injection**
- Updated API constructor to include `PublishingService`
- Enhanced `APIHandlers` struct with publishing service dependency
- Updated all route configurations in `cmd/operion-api/api.go`

### 🔧 **Standardized Error Handling System**

This PR introduces a robust, type-safe error handling system to replace fragile string-based error checking:

#### **Key Improvements**
- **Standardized Error Constants**: `ErrWorkflowNotFound`, `ErrPublishedWorkflowNotFound`, etc.
- **Contextual Error Wrappers**: Rich error information with operation context
- **Type-Safe Error Checking**: `IsWorkflowNotFound(err)`, `IsPublishedWorkflowNotFound(err)` functions
- **Consistent HTTP Status Mapping**: Proper 404/409/500 status codes based on error type

#### **Before vs After**
```go
// Before (Fragile)
if strings.Contains(err.Error(), "not found") {
    return notFound(c, "Workflow not found")
}

// After (Type-Safe)  
if persistence.IsWorkflowNotFound(err) {
    return notFound(c, "Workflow not found")
}
```

#### **Unit Tests**
- ✅ Complete test coverage for all new handlers
- ✅ Request/response DTO validation testing
- ✅ Error handling and edge case testing
- ✅ Owner-based filtering functionality testing
- ✅ Status change operations testing

#### **Integration Tests**
- ✅ End-to-end workflow CRUD lifecycle testing with Testcontainers
- ✅ PostgreSQL integration testing with real database
- ✅ Publishing workflow integration testing
- ✅ Validation error testing with real HTTP requests

### 🚀 **API Endpoints Ready for Production**

```bash
# Complete CRUD Operations
POST   /workflows                                    # Create workflow
GET    /workflows                                    # List all workflows
GET    /workflows?owner_id=user123                   # List workflows by owner
GET    /workflows/:id                                # Get workflow by ID
PATCH  /workflows/:id                                # Update workflow  
DELETE /workflows/:id                                # Delete workflow

# Status Management
POST   /workflows/:id/publish                        # Publish workflow
POST   /workflows/groups/:groupId/create-draft       # Create draft from published
```

### ✅ **Quality Assurance**

- **Code Formatting** - `go fmt ./...` applied across all files
- **Static Analysis** - `go vet ./...` passed successfully
- **Build Verification** - `make build` completed for all services
- **Comprehensive Testing** - `make test` passed with 100% success rate
- **Race Condition Testing** - All tests pass with race detection
- **Struct Tag Alignment** - Proper tag formatting per project standards

### 📊 **Test Results**

- **Total Test Suites**: 9/9 completed ✅
- **Build Status**: All services build successfully ✅
- **Code Coverage**: Comprehensive unit and integration tests ✅
- **Static Analysis**: All checks passed ✅
- **Error Handling**: Standardized and robust ✅

🤖 Generated with [Claude Code](https://claude.ai/code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: cmd/operion-source-manager/manager.go [hunk 1]
```diff
@@ -287,14 +287,17 @@ func (spm *ProviderManager) updateTriggersWithSourceIDs(ctx context.Context, tri
 	spm.logger.Info("Updating triggers with source IDs", "mapping_count", len(triggerToSourceMap))
 
 	// Get all workflows to find and update triggers
-	workflows, err := spm.persistence.WorkflowRepository().GetAll(ctx)
+	result, err := spm.persistence.WorkflowRepository().ListWorkflows(ctx, persistence.ListWorkflowsOptions{
+		Limit:        1000, // Get all workflows for source ID management
+		IncludeNodes: true, // Need nodes for trigger processing
```
Comment (renancaju) [inline cmd/operion-source-manager/manager.go:292]: We will need to revisit this in the future... The source manager only does this because there was no way to create a complete workflow via endpoints. This was the way found to read the json files and compute the sources to be handled. When all of the endpoints are created, we can remove this.

File: cmd/operion-source-manager/manager.go [hunk 2]
```diff
@@ -369,7 +372,13 @@ func (spm *ProviderManager) executeProviderLifecycle(ctx context.Context, lifecy
...
 
 	// Step 2: Configure with current workflows
-	workflows, err := spm.persistence.WorkflowRepository().GetAll(ctx)
+	publishedStatus := models.WorkflowStatusPublished
+
+	result, err := spm.persistence.WorkflowRepository().ListWorkflows(ctx, persistence.ListWorkflowsOptions{
+		Status:       &publishedStatus, // Only get published workflows for source configuration
+		Limit:        1000,             // Get all published workflows
+		IncludeNodes: true,             // Need nodes for source configuration
+	})
```
Comment (renancaju) [inline cmd/operion-source-manager/manager.go:381]: Similar comment to the previous one. This will soon be removed.

File: pkg/persistence/file/workflow.go [hunk 3]
```diff
@@ -24,8 +26,32 @@ func NewWorkflowRepository(root string) *WorkflowRepository {
...
+
+	// Validate sort parameters against allowlist (security)
+	allowedSorts := map[string]bool{
+		"created_at": true,
+		"updated_at": true,
+		"name":       true,
+	}
+	if !allowedSorts[opts.SortBy] {
+		return nil, fmt.Errorf("invalid sort field: %s", opts.SortBy)
+	}
```
Comment (renancaju) [inline pkg/persistence/file/workflow.go:52]: Use proper repository errors

File: pkg/persistence/file/workflow.go [hunk 4]
```diff
@@ -34,21 +60,101 @@ func (wr *WorkflowRepository) GetAll(ctx context.Context) ([]*models.Workflow, e
...
+	// Load all workflows
+	allWorkflows := make([]*models.Workflow, 0, len(jsonFiles))
 	for _, file := range jsonFiles {
-		workflow, err := wr.GetByID(ctx, file[:len(file)-5])
+		workflowID := file[:len(file)-5] // Remove .json extension
+
+		workflow, err := wr.GetByID(ctx, workflowID)
 		if err != nil {
-			return nil, err
+			return nil, fmt.Errorf("failed to load workflow %s: %w", workflowID, err)
```
Comment (renancaju) [inline pkg/persistence/file/workflow.go:77]: Use proper repository errors

File: pkg/persistence/postgresql/postgres.go [hunk 5]
```diff
@@ -86,32 +85,6 @@ func (p *Persistence) HealthCheck(ctx context.Context) error {
...
-func (p *Persistence) DeleteWorkflow(ctx context.Context, id string) error {
-	return p.workflowRepo.Delete(ctx, id)
-}
-
-// WorkflowTriggersBySourceEventAndProvider finds triggers by source ID, event type, and provider ID.
-// Deprecated: Use NodeRepository.FindTriggerNodesBySourceEventAndProvider instead.
-func (p *Persistence) WorkflowTriggersBySourceEventAndProvider(ctx context.Context, sourceID, eventType, providerID string, status models.WorkflowStatus) ([]*models.TriggerNodeMatch, error) {
-	return p.nodeRepo.FindTriggerNodesBySourceEventAndProvider(ctx, sourceID, eventType, providerID, status)
-}
-
```
Comment (renancaju) [inline pkg/persistence/postgresql/postgres.go:114]: I have moved this to the repository

File: pkg/persistence/postgresql/workflow.go [hunk 6]
```diff
@@ -24,61 +26,311 @@ func NewWorkflowRepository(db *sql.DB, logger *slog.Logger) *WorkflowRepository
...
+	if opts.SortBy != "" {
+		validSortColumns := map[string]string{
+			"created_at": "created_at",
+			"updated_at": "updated_at",
+			"name":       "name",
+		}
+
+		validColumn, exists := validSortColumns[opts.SortBy]
+		if !exists {
+			return "", nil, fmt.Errorf("invalid sort column: %s", opts.SortBy)
```
Comment (renancaju) [inline pkg/persistence/postgresql/workflow.go]: Use proper repository errors
